regression/regression_utils.py

```diff
@@ -31,6 +32,38 @@
 """
 
 
+class RetryTestsMetaclass(type):
+
+    NUM_RETRIES = 2
+    FLAKY_ERROR_CLASSES = (AssertionError,)
```

Comment:
@tseaver D'oh! I forgot I had an example in the initial bug. WDYT about adding

```
from gcloud.storage.exceptions import NotFound
...
     NUM_RETRIES = 2
    FLAKY_ERROR_CLASSES = (AssertionError, NotFound)
```

As for failing assertions, we sometimes have cases (https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/46219691) where eventual consistency causes queries not to return what we expect, so a length of results assertion fails.

These failures "shouldn't" occur since we use transactions (or query for data that is weeks old, having been inserted via `populate_datastore.py`) but we see them often enough.